### ISSUE TITLE
64-bit number encoding/decoding in msgpack does not work

### ISSUE THREAD

**Issue by pingyuusa** [msgpack/msgpack-node#37]
Due to the limitation of javascript the way numbers are store, 64 bit integers can not represented correctly.

msgpack decodes a 64 bit number as ordinary javascript numeric type which is incorrect and looses precision.

n = "18446464814936021036"
x = [ 0xcf, 0xff, 0xff, 0x02, 0x04, 0x00, 0x00, 0xd8, 0x2c ]
a = msgpack.unpack(x)
a == 1844646481493602000 // not equal to n

unittest for this is incorrect too.

Need an alternate way to encode/decode 64 bit numbers to/from msgpack format

**Comment by bin-y** [comment]
64-bit numbers should be mapped to [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt)

**Comment by bin-y** [comment]
ping @pgriess @godsflaw

**Comment by godsflaw** [comment]
Yeah, this should be solved with one of the BigInt packages.  I'm very busy, but would be happy to take a PR.

**Comment by bin-y** [comment]
hi, I did write some code to implement this for my own use. But my implemention used BigInt type directly and ignored backward compatibility. Due to the compatibility issue, I think it won't be a good PR, but it could be a reference if you need. Here is my commit https://github.com/bin-y/msgpack-node/commit/bccf8a0e62b8b9075a205c747f5850c3414dd313

**Comment by enochgroot** [comment]
Decoding 64-bit integers as BigInt (or string) would change 2.0.0 unpack results for existing callers. Out of scope for 2.0.0 (#25687); belongs in a 3.0 discussion if anyone wants to take it on.

**Comment by enochgroot** [comment]
Repro on master `e04c9b55`, Node v20.20.2:

```
unpack(Buffer.from('cfffffffffffffffff','hex')) → 18446744073709552000  // Number, not 2^64-1
pack(18446744073709551615n) → throws "cannot pack object"
```

`src/msgpack.cc` still decodes uint64/int64 through `v8::Number` (legacy). Returning BigInt would be a 3.0 API. Leaving open.
